internal/pkg/agent/perms/windows.go

```diff
@@ -121,13 +119,50 @@ func FixPermissions(topPath string, opts ...OptFunc) error {
 }
 
 func takeOwnership(name string, owner *windows.SID, group *windows.SID) error {
-	return api.SetNamedSecurityInfo(
+	return windows.SetNamedSecurityInfo(
 		name,
-		api.SE_FILE_OBJECT,
-		api.OWNER_SECURITY_INFORMATION|api.GROUP_SECURITY_INFORMATION,
+		windows.SE_FILE_OBJECT,
+		windows.OWNER_SECURITY_INFORMATION|windows.GROUP_SECURITY_INFORMATION,
 		owner,
 		group,
-		0,
-		0,
+		nil,
+		nil,
+	)
+}
+
+func grantSid(accessPermissions uint32, sid *windows.SID) windows.EXPLICIT_ACCESS {
+	return windows.EXPLICIT_ACCESS{
+		AccessPermissions: windows.ACCESS_MASK(accessPermissions),
+		AccessMode:        windows.GRANT_ACCESS,
+		Inheritance:       windows.SUB_CONTAINERS_AND_OBJECTS_INHERIT,
+		Trustee: windows.TRUSTEE{
+			TrusteeForm:  windows.TRUSTEE_IS_SID,
+			TrusteeValue: windows.TrusteeValueFromSID(sid),
+		},
+	}
+}
+
+func apply(name string, inherit bool, entries ...windows.EXPLICIT_ACCESS) error {
+	acl, err := windows.ACLFromEntries(
+		entries,
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+	var secInfo windows.SECURITY_INFORMATION
+	if !inherit {
+		secInfo = windows.PROTECTED_DACL_SECURITY_INFORMATION
+	} else {
+		secInfo = windows.UNPROTECTED_DACL_SECURITY_INFORMATION
+	}
+	return windows.SetNamedSecurityInfo(
```

Comment:
I just merged https://github.com/elastic/elastic-agent/pull/5317 which checked that `acl.Apply` did not return `syscall.Errno(0)`
I think you would need to make a similar check here, I'm not sure if `windows.ACLFromEntries` would need the same check or not